### ISSUE TITLE
Predict contribution gives wrong number of contributions in R

### ISSUE THREAD

**Issue by ThomasWolf0701** [microsoft/LightGBM#3487]
_ ran the following code:_

data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)
data(agaricus.test, package = "lightgbm")
test <- agaricus.test
dtest <- lgb.Dataset.create.valid(dtrain, test$data, label = test$label)
params <- list(objective = "regression", metric = "l2")
valids <- list(test = dtest)
model <- lgb.train(
  params = params
  , data = dtrain
  , nrounds = 5L
  , valids = valids
  , min_data = 1L
  , learning_rate = 1.0
)

Then tried:
contribs <- predict(model, test$data,predcontrib = TRUE)

Now I checked if the dimension match:
dim(contribs)
1611 127

but 
dim(test$data)
1611 126

So seemingly contribs "invented" feature, thats not in the original data. Also can´t check the reason as contribs has no rownames/colnames

**Comment by jameslamb** [comment]
Hi @ThomasWolf0701 , thanks for using `{lightgbm}` and for providing this reproducible example. I'll look into this later today and get back to you.

**Comment by ThomasWolf0701** [comment]
> 
> 
> Hi @ThomasWolf0701 , thanks for using `{lightgbm}` and for providing this reproducible example. I'll look into this later today and get back to you.

Was thinking one could be some kind of intercept but this is nor mentioned anywhere.

**Comment by StrikerRUS** [comment]
@ThomasWolf0701 Hi!

For the additional column in predictions please refer to 

> produces `#features + 1` values where the last value is the expected value of the model output over the training data
> Note: unlike the shap package, with `predict_contrib` we return a matrix with an extra column, where the last column is the expected value
  https://lightgbm.readthedocs.io/en/latest/Parameters.html#predict_contrib

and https://github.com/microsoft/LightGBM/issues/1969#issuecomment-480580776

**Comment by ThomasWolf0701** [comment]
> 
> 
> @ThomasWolf0701 Hi!
> 
> For the additional column in predictions please refer to
> 
> > produces `#features + 1` values where the last value is the expected value of the model output over the training data
> > Note: unlike the shap package, with `predict_contrib` we return a matrix with an extra column, where the last column is the expected value
> > https://lightgbm.readthedocs.io/en/latest/Parameters.html#predict_contrib
> 
> and [#1969 (comment)](https://github.com/microsoft/LightGBM/issues/1969#issuecomment-480580776)

Many thanks for the information, would still be great to mention this in the R documentation/help

**Comment by jameslamb** [comment]
thanks @StrikerRUS ! I actually forgot that's how it works :grinning: 

@ThomasWolf0701 I'll look for the right place to include this in the R docs that wouldn't be a duplication of our existing Parameters. In general, all parameters at https://lightgbm.readthedocs.io/en/latest/Parameters.html are passed through from the R package to the core LightGBM C++ library, and you should expect the behavior of the R package to be defined by what is described there.
